docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc

```diff
@@ -0,0 +1,328 @@
+The init container is already defined, so uncomment the respective section:
+
+["source", "yaml", subs="attributes"]
+----
+initContainers:
+- name: k8s-templates-downloader
+  image: busybox:1.28
+  command: ['sh']
+  args:
+    - -c
+    - >-
+      mkdir -p /etc/elastic-agent/inputs.d &&
+      wget -O - https://github.com/elastic/elastic-agent/archive/{branch}.tar.gz | tar xz -C /etc/elastic-agent/inputs.d --strip=5 "elastic-agent-main/deploy/kubernetes/elastic-agent-standalone/templates.d"
+  volumeMounts:
+    - name: external-inputs
+      mountPath: /etc/elastic-agent/inputs.d
+----
+
+[discrete]
+=== Hints autodiscovery: Example (standalone agent)
+
+Having the hints mechanism enabled allows users deploying Pods on the cluster to automatically turn on Elastic's
+monitoring for their Pod, at the time of deploying their Pods.
+For example if we want deploy a Redis Pod on the cluster and want to automatically enable Elastic monitoring for it
+we need to add the proper hints as annotations on the Pod:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: redis
+  annotations:
+    co.elastic.hints/package: redis
+    co.elastic.hints/data_streams: info, key, keyspace
+    co.elastic.hints/host: '${kubernetes.pod.ip}:6379'
+    co.elastic.hints/info.period: 5s
+  labels:
+    k8s-app: redis
+    app: redis
+spec:
+  containers:
+  - image: redis
+    imagePullPolicy: IfNotPresent
+    name: redis
+    ports:
+    - name: redis
+      containerPort: 6379
+      protocol: TCP
+----
+
+After deploying this Pod the data will start flowing in automatically. Note that at this point all the assets
+(dashboards, ingest pipelines etc) related to the Redis integration are not installed and one would need to
+explicitly install them through Kibana.
+
+[discrete]
+=== Hints autodiscovery: Troubleshooting (standalone agent)
+
+When things do not work as expected one would need to troubleshoot their setup. In that case many things can be wrong
+but here we provide some directions to speed up your investigation:
+
+. Exec inside an Agent's Pod and run inspect command to verify how inputs are constructed dynamically:
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+./elastic-agent inspect -c /etc/elastic-agent/agent.yml output -o default -v -d "*"
+------------------------------------------------
++
+Specifically check how the inputs are being populated.
+
+. Check the logs of Elastic Agent:
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+tail -f /etc/elastic-agent/data/logs/elastic-agent-*.ndjson
+------------------------------------------------
++
+Verify that the hints feature is enabled in the config and look for hints' related logs like:
+"Generated hints mappings are ..."
+In this, you can find what are the mappings that are extracted out of the annotations and have an indicator
+if these values are able to populate a specific input.
```

Comment:
Here's my attempt at an edit, but without seeing the log entries, the details here are a bit hard to parse.

```suggestion
Verify that the hints feature is enabled in the config and look for hints-related logs like:
"Generated hints mappings are ..."
In these logs, you can find the mappings that are extracted out of the annotations and determine
if the values are able to populate a specific input.
```